tokio-trace-fmt/src/span.rs

```diff
@@ -337,10 +345,13 @@ impl Store {
 }
 
 impl Data {
-    pub(crate) fn new(metadata: &Metadata) -> Self {
+    pub(crate) fn new<N>(metadata: &Metadata) -> Self
+    where
+        N: for<'a> ::NewVisitor<'a>,
+    {
         Self {
             name: metadata.name(),
-            parent: Context::current(),
+            parent: Context::<N>::current(),
```

Comment:
I think we should probably just move `current` out of the `Context` type and make it a free function, rather than having to make `Data::new` generic over `N`. There's really no reason for `current` to be defined by `Context`...